src/doc/trpl/guessing-game.md

```diff
@@ -0,0 +1,991 @@
+$ cargo build
+   Compiling guessing_game v0.1.0 (file:///home/you/projects/guessing_game)
+src/main.rs:28:21: 28:35 error: mismatched types:
+ expected `&collections::string::String`,
+    found `&_`
+(expected struct `collections::string::String`,
+    found integral variable) [E0308]
+src/main.rs:28     match guess.cmp(&secret_number) {
+                                   ^~~~~~~~~~~~~~
+error: aborting due to previous error
+Could not compile `guessing_game`.
+```
+
+Whew! This is a big error. The core of it is that we have ‘mismatched types’.
+Rust has a strong, static type system. However, it also has type inference.
+When we wrote `let guess = String::new()`, Rust was able to infer that `guess`
+should be a `String`, and so it doesn’t make us write out the type. And with
+our `secret_number`, there are a number of types which can have a value
+between one and a hundred: `i32`, a thirty-two-bit number, or `u32`, an
+unsigned thirty-two-bit number, or `i64`, a sixty-four-bit number. Or others.
+So far, that hasn’t mattered, and so Rust defaults to an `i32`. However, here,
+Rust doesn’t know how to compare the `guess` and the `secret_number`. They
+need to be the same type. Ultimately, we want to convert the `String` we
+read as input into a real number type, for comparison. We can do that
+with three more lines. Here’s our new program:
+
+```rust,ignore
+extern crate rand;
+
+use std::io;
+use std::cmp::Ordering;
+use rand::Rng;
+
+fn main() {
+    println!("Guess the number!");
+
+    let secret_number = rand::thread_rng().gen_range(1, 101);
+
+    println!("The secret number is: {}", secret_number);
+
+    println!("Please input your guess.");
+
+    let mut guess = String::new();
+
+    io::stdin().read_line(&mut guess)
+        .ok()
+        .expect("failed to read line");
+
+    let guess: u32 = guess.trim().parse()
+        .ok()
+        .expect("Please type a number!");
+
+    println!("You guessed: {}", guess);
+
+    match guess.cmp(&secret_number) {
+        Ordering::Less    => println!("Too small!"),
+        Ordering::Greater => println!("Too big!"),
+        Ordering::Equal   => println!("You win!"),
+    }
+}
+```
+
+The new three lines:
+
+```rust,ignore
+    let guess: u32 = guess.trim().parse()
+        .ok()
+        .expect("Please type a number!");
+```
+
+Wait a minute, I thought we already had a `guess`? We do, but Rust allows us
+to ‘shadow’ the previous `guess` with a new one. This is often used in this
+exact situation, where `guess` starts as a `String`, but we want to convert it
+to an `i32`. Shadowing lets us re-use the `guess` name, rather than forcing us
+to come up with two unique names like `guess_str` and `guess`, or something
+else.
+
+We bind `guess` to an expression that looks like something we wrote earlier:
+
+```rust,ignore
+guess.trim.parse()
+```
+
+Followed by an `ok().expect()` invocation. Here, `guess` refers to the old
+`guess`, the one that was a `String` with our input in it. The `trim`()`
```

Comment:
ah! thank you! I was getting weird rendering, but for the life of me, couldn't figure out hwere.